src/transaction/types/input.rs

```diff
@@ -260,7 +263,10 @@ impl io::Write for Input {
                 let (size, predicate_data, _) = bytes::restore_raw_bytes(buf, predicate_data_len)?;
                 n += size;
 
-                let utxo_id = utxo_id.into();
+                let utxo_id = UtxoId {
```

Comment:
We need constructors to encapsulate the behavior of `UtxoId` instead of coupling the code here.